java/arcs/core/storage/testutil/FakeActiveStore.kt

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright 2021 Google LLC.
+ *
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ *
+ * Code distributed by Google as part of this project is also subject to an additional IP rights
+ * grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+
+package arcs.core.storage.testutil
+
+import arcs.core.crdt.CrdtData
+import arcs.core.crdt.CrdtOperation
+import arcs.core.storage.ActiveStore
+import arcs.core.storage.ProxyCallback
+import arcs.core.storage.ProxyMessage
+import arcs.core.storage.StoreOptions
+
+/** [FakeActiveStore] is used to test exception handling for callback registration. */
```

Comment:
This is a little bit strange. Testing what happens when an exception is thrown is totally fine of course, but having that happen as the default behaviour for a fake is quite unexpected (particularly when the class name `FakeActiveStore` gives no indication that it's specifically for testing exceptions).

There's a couple of other ideas you could consider:

1. Add a nullable `exception` field to this class, and have the `on` method either throw the exception it sees there, or return a default int
2. Make this an open class, make the default behaviour for the `on` method to just return some value (e.g. `0`), and then subclass this fake class in the unit test itself to do the exceptional behaviour.

My preference would be for option 2. 

... And in fact we already have a `NoopActiveStore` class that allows you to do that. You can see an example of overriding this very method in `StorageChannelImplTest.kt`

Reply:
Awesome, I'm happy to use something that already exists. Thanks for the suggestions!